crates/bevy_winit/src/system.rs

```diff
@@ -146,6 +150,14 @@ pub(crate) fn despawn_windows(
             closed_events.send(WindowClosed { window });
         }
     }
+
+    // On macOS, when exiting, we need to tell the rendering thread the windows are about to
+    // close to ensure that they are dropped on the main thread. Otherwise, the app will hang.
+    for _ in exit_events.read() {
+        for window in window_entities.iter() {
+            closing_events.send(WindowClosing { window });
+        }
+    }
```

Comment:
This will send duplicate `WindowClosing` events if multiple exit events are raised in a single update. You could use `EventReader::is_empty` instead of the for loop to fix this. 
```suggestion
    if !exit_events.is_empty() {
        exit_events.clear();
        for window in window_entities.iter() {
            closing_events.send(WindowClosing { window });
        }
    }
```

Reply:
good call, sending multiple exit events is a little weird but better to be safe! ty